dvc/api.py

```diff
@@ -69,3 +72,48 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, params=None, repo=None, rev=None):
+    # 1. Read artifacts.yaml
+    # 2. Pull dependencies
+    # 3. Get the call and parameters
+    # 4. Invoke the call with the given parameters
+    # 5. Return the result
+
+    with _make_repo(repo, rev=rev) as _repo:
+        artifacts_path = os.path.join(_repo.root_dir, "artifacts.yaml")
+
+        with open(artifacts_path, "r") as fobj:
+            artifacts = ruamel.yaml.safe_load(fobj.read()).get("artifacts")
+            artifact = next(x for x in artifacts if x.get("name") == name)
+
+        outs = [
+            _repo.find_outs_by_path(os.path.join(_repo.root_dir, dep))[0]
+            for dep in artifact.get("deps")
+        ]
+
+        with _repo.state:
+            for out in outs:
+                _repo.cloud.pull(out.get_used_cache())
+                out.checkout()
+
+        previous_dir = os.path.abspath(os.curdir)
+
+        os.chdir(_repo.root_dir)
```

Comment:
@Suor , this would require the user to always us absolute path names:

The test case I wrote:

```python
def add_to_num(x): return x + int(open('number').read())
```

Is not passing since `number` is living on the external repository but code is being executed in the local(?) repository.

Reply:
Is this intended? Or we should support both?

Reply:
Ah, I see. Let's chdir. However, we might have some of issues with that:
- not thread safe
- import will pollute `sys.modules`
- if there is a name clash within `sys.modules` we might get a weird error
  
Let's ignore that for now.

Reply:
Still use `try/finally` or a context manager.

Reply:
Let's separate this import/instantiate code a separate func though.